plugins/Tour/Engagement/ChallengeCustomLogo.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\Tour\Engagement;
+
+use Piwik\Piwik;
+use Piwik\Plugins\CoreAdminHome\CustomLogo;
+use Piwik\Plugins\Tour\Dao\DataFinder;
+use Piwik\Url;
+
+class ChallengeCustomLogo extends Challenge
+{
+    /**
+     * @var DataFinder
+     */
+    private $finder;
+
+    /**
+     * @var null|bool
+     */
+    private $completed = null;
+
+    public function __construct(DataFinder $dataFinder)
+    {
+        $this->finder = $dataFinder;
+    }
+
+    public function getName()
+    {
+        return Piwik::translate('Tour_UploadLogo');
+    }
+
+    public function getDescription()
+    {
+        return Piwik::translate('CoreAdminHome_CustomLogoHelpText');
+    }
+
+    public function getId()
+    {
+        return 'custom_logo';
+    }
+
+    public function isCompleted()
+    {
+        if (!isset($this->completed)) {
+            $logo = new CustomLogo();
+            $this->completed = $logo->isEnabled();
+        }
+        return $this->completed;
+    }
+
+    public function getUrl()
+    {
+        return 'index.php' . Url::getCurrentQueryStringWithParametersModified(array('module' => 'CoreAdminHome', 'action' => 'generalSettings', 'widget' => false)) . '#useCustomLogo';
```

Comment:
Angular turns '#useCustomLogo' into '#/useCustomLogo', so it doesn't go to the correct section.

Reply:
Any thoughts on how to fix this?

Reply:
Hacky fix might be on page load to check if the hash has no query parameter characters (ie, only the leading '/'), check if there's an element w/ that string as an ID, and if so scroll to it. Not sure how else to do it since the hash is used by angular...

Could use a special query parameter too instead of the hash, like, `scrollTo=myelement`.

Reply:
> Could use a special query parameter too instead of the hash, like, scrollTo=myelement.

This might be sometimes tricky when other elements are only loaded later through XHR and then rendered later by the time we already long processed `scrollTo` parameter. 

Seems `#/#useCustomLogo` works. Can you confirm?